api/envoy/config/core/v3/socket_option.proto

```diff
@@ -51,6 +51,17 @@ message SocketOption {
     STATE_LISTENING = 2;
   }
 
+  enum SocketType {
```

Comment:
OOC do you think this will need per-type specific config?
For example, if STREAM is used, then there could be a protocol attached to it.
If so, then I suggest converting to messages rather than using enums.
This is just a general comment, if you think it is ok, this can be kept as an enum.

Reply:
Good idea. Done, updated to use messages.